.github/workflows/main.yaml

```diff
@@ -27,14 +32,45 @@ jobs:
         uses: actions/setup-node@v2
         with:
           node-version: "14"
+          cache: "yarn"
+          cache-dependency-path: |
+            client/yarn.lock
 
       - name: Install NPM dependencies
-        run: npm install
+        run: yarn
+        working-directory: client
 
       - name: Client linting
-        run: npm install && npm run prettier:check
+        run: yarn run prettier:check
         working-directory: client
 
+  brownie-tests:
+    name: Brownie tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Cache Compiler Installations
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.solcx
```

Comment:
curious does this preserve the build artefacts from previous compilations?

Reply:
Actually no, this is for solidity compiler. But we cache built artifacts on the origin-dollar repo since we use that a lot for fork tests. Can do the same optimisations here too